File: Polimorfismo/Ejercicio9/Java/Minecraft.java
class BloqueCofre {
    private int capacidad;
    private int resistencia;
    private String tipo;
    public BloqueCofre(int capacidad, int resistencia, String tipo) {
        this.capacidad = capacidad;
        this.resistencia = resistencia;
        this.tipo = tipo;
    }
    public void accionar() {
        System.out.println("Se abre el cofre.");
    }
    public void romper() {
        System.out.println("El cofre se rompió y soltó objetos.");
    }
    public void colocar(String colocar) {
        System.out.println("el bloque se puso en " + colocar);
    }
}
class BloqueTnt {
    private String tipo;
    private int daño;
    public BloqueTnt(String tipo, int daño) {
        this.tipo = tipo;
        this.daño = daño;
    }
    public void accionar() {
        System.out.println("La TNT se encendió.");
    }
    public void romper() {
        System.out.println("¡BOOM! La TNT explotó.");
    }
    public void colocar(String colocar) {
        System.out.println("el bloque se puso en " + colocar);
    }
}
class BloqueHorno {
    private String color;
    private int capacidadDeComida;
    public BloqueHorno(String color, int capacidadDeComida) {
        this.color = color;
        this.capacidadDeComida = capacidadDeComida;
    }
    public void accionar() {
        System.out.println("El horno se enciende para cocinar.");
    }
    public void romper() {
        System.out.println("El horno se rompió.");
    }
    public void colocar(String colocar) {
        System.out.println("el bloque se puso en " + colocar);
    }
}

File: Polimorfismo/Ejercicio9/Python/Minecraft.py
class BloqueCofre:
    def __init__(self, capacidad, resistencia, tipo):
        self.capacidad = capacidad
        self.resistencia = resistencia
        self.tipo = tipo
    def accionar(self):
        print("Se abre el cofre.")
    def romper(self):
        print("El cofre se rompió y soltó objetos.")
    def colocar(self, colocar):
        print("el bloque se puso en", colocar)
class BloqueTnt:
    def __init__(self, tipo, daño):
        self.tipo = tipo
        self.daño = daño
    def accionar(self):
        print("La TNT se encendió.")
    def romper(self):
        print("¡BOOM! La TNT explotó.")
    def colocar(self, colocar):
        print("el bloque se puso en", colocar)
class BloqueHorno:
    def __init__(self, color, capacidadDeComida):
        self.color = color
        self.capacidadDeComida = capacidadDeComida
    def accionar(self):
        print("El horno se enciende para cocinar.")
    def romper(self):
        print("El horno se rompió.")
    def colocar(self, colocar):
        print("el bloque se puso en", colocar)
b1 = BloqueCofre()
b2 = BloqueTnt()
b3 = BloqueHorno()
b1.accionar()
b2.accionar()
b3.accionar()
b1.colocar("el suelo")
b2.colocar("la pared")
b1.romper()
b2.romper()
b3.romper()